chaste_codegen/data/tests/chaste_reference_models/BE/Shannon2004_with_modifiers.cpp

```diff
@@ -0,0 +1,2159 @@
+//! @file
+//!
+//! This source file was generated from CellML by chaste_codegen version 0.0.1
+//!
+//! Model: shannon_2004
+//!
+//! Processed by chaste_codegen: https://github.com/ModellingWebLab/chaste-codegen
+//!     (translator: chaste_codegen, model type: normal)
+//! on (date omitted as unimportant)
+//!
+//! <autogenerated>
+
+#include "Shannon2004_with_modifiers.hpp"
+#include <cmath>
+#include <cassert>
+#include <memory>
+#include "Exception.hpp"
+#include "OdeSystemInformation.hpp"
+#include "RegularStimulus.hpp"
+#include "HeartConfig.hpp"
+#include "IsNan.hpp"
+#include "MathsCustomFunctions.hpp"
+#include "CardiacNewtonSolver.hpp"
+
+    boost::shared_ptr<RegularStimulus> CellShannon2004FromCellML::UseCellMLDefaultStimulus()
+    {
+        // Use the default stimulus specified by CellML metadata
+        const double var_chaste_interface__cell__stim_amplitude = 9.5 * HeartConfig::Instance()->GetCapacitance(); // microA_per_microF
+        const double var_chaste_interface__cell__stim_duration = 5.0; // millisecond
+        const double var_chaste_interface__cell__stim_period = 1000.0; // millisecond
+        const double var_chaste_interface__cell__stim_start = 100.0; // millisecond
+        boost::shared_ptr<RegularStimulus> p_cellml_stim(new RegularStimulus(
+                -fabs(var_chaste_interface__cell__stim_amplitude),
+                var_chaste_interface__cell__stim_duration,
+                var_chaste_interface__cell__stim_period,
+                var_chaste_interface__cell__stim_start
+                ));
+        mpIntracellularStimulus = p_cellml_stim;
+        return p_cellml_stim;
+    }
+    double CellShannon2004FromCellML::GetIntracellularCalciumConcentration()
+    {
+        return mStateVariables[1];
+    }
+    CellShannon2004FromCellML::CellShannon2004FromCellML(boost::shared_ptr<AbstractIvpOdeSolver> /* unused; should be empty */, boost::shared_ptr<AbstractStimulusFunction> pIntracellularStimulus)
+        : AbstractBackwardEulerCardiacCell<26>(
```

Comment:
This isn't a modifier-capable base class; don't think this would run.

Reply:
From `the Abstract...WithModifiers.cpp` file, the only supported instances are:
```cpp

// Explicit Instantiation
template class AbstractCardiacCellWithModifiers<AbstractCardiacCell>;
#ifdef CHASTE_CVODE
template class AbstractCardiacCellWithModifiers<AbstractCvodeCell>;
template class AbstractCardiacCellWithModifiers<AbstractCvodeCellWithDataClamp>;
#endif
```

Reply:
The approach used will certainly only work if the cell class has the same constructor signature, which Backward Euler does not.

Reply:
uups I hadn't realised